src/Couchbase.Lite.Shared/Serialization/FleeceMutableArray.cs

```diff
@@ -39,12 +48,16 @@ internal sealed unsafe class MArray : MCollection, IList<object>
 
         #region Properties
 
-        public FLArray* BaseArray { get; private set; }
+        private FLMutableArray* ar { get; set; }
+
+        private FLArray* BaseArray => (FLArray*)ar;
```

Comment:
Upon looking at this, I think it might be better to have one `_flArray` variable of type `FLArray *` and simply cast as needed.  I thought this would be two classes, but it is just one (which is better).